diff --git a/node_modules/@mapbox/togeojson/togeojson.js b/node_modules/@mapbox/togeojson/togeojson.js
index b06ad53..572b1bd 100644
--- a/node_modules/@mapbox/togeojson/togeojson.js
+++ b/node_modules/@mapbox/togeojson/togeojson.js
@@ -85,7 +85,7 @@ var toGeoJSON = (function() {
         serializer = new XMLSerializer();
     // only require xmldom in a node environment
     } else if (typeof exports === 'object' && typeof process === 'object' && !process.browser) {
-        serializer = new (require('xmldom').XMLSerializer)();
+        serializer = require('xmldom');
     }
     function xml2str(str) {
         // IE9 will create a new XMLSerializer but it'll crash immediately.
