diff --git a/node_modules/@react-native-mapbox-gl/maps/ios/RCTMGL/RCTMGLLocationManager.m b/node_modules/@react-native-mapbox-gl/maps/ios/RCTMGL/RCTMGLLocationManager.m
index cc38935..5ad871a 100644
--- a/node_modules/@react-native-mapbox-gl/maps/ios/RCTMGL/RCTMGLLocationManager.m
+++ b/node_modules/@react-native-mapbox-gl/maps/ios/RCTMGL/RCTMGLLocationManager.m
@@ -51,7 +51,7 @@ - (void)start:(CLLocationDistance)minDisplacement
     }
     
     dispatch_async(dispatch_get_main_queue(), ^{
-        [locationManager requestWhenInUseAuthorization];
+        [locationManager requestAlwaysAuthorization];
         [locationManager startUpdatingLocation];
         [locationManager startUpdatingHeading];
         [locationManager setDistanceFilter:(minDisplacement)];
