diff --git a/node_modules/react-native/React/Views/RefreshControl/RCTRefreshControl.m b/node_modules/react-native/React/Views/RefreshControl/RCTRefreshControl.m
index 38f2ab5..087d17c 100644
--- a/node_modules/react-native/React/Views/RefreshControl/RCTRefreshControl.m
+++ b/node_modules/react-native/React/Views/RefreshControl/RCTRefreshControl.m
@@ -97,7 +97,13 @@ - (void)endRefreshingProgrammatically
   UIScrollView *scrollView = self.scrollView;
   if (scrollView && _refreshingProgrammatically && scrollView.contentOffset.y < -scrollView.contentInset.top) {
     UInt64 endRefreshingTimestamp = _currentRefreshingStateTimestamp;
-    CGPoint offset = {scrollView.contentOffset.x, -scrollView.contentInset.top};
+    UIEdgeInsets contentInset;;
+    if (@available(iOS 11.0, *)) {
+        contentInset = scrollView.adjustedContentInset;
+    } else {
+        contentInset = scrollView.contentInset;
+    }
+    CGPoint offset = {scrollView.contentOffset.x, -contentInset.top};
     [UIView animateWithDuration:0.25
         delay:0
         options:UIViewAnimationOptionBeginFromCurrentState
